File: 2023/day1/main.py
"""
Find the elf with the most calories and return it given the text document input
"""
import os

dir_path = os.path.dirname(os.path.realpath(__file__))
lines = []
maxCalories = 0
currentLine = 0

with open(f"{dir_path}/input.txt") as plainInput:
    for line in plainInput.readlines():
        a = line.strip()
        lines.append(a)

while currentLine < len(lines):
    elfCalories = 0

    while lines[currentLine] != "" and currentLine + 1 < len(lines):
        elfCalories += int(lines[currentLine])
        currentLine += 1

    maxCalories = max(elfCalories, maxCalories)
    currentLine += 1

print(maxCalories)

File: 2023/day1/rust/src/main.rs
/* 
*** problem ***:
given input.txt file
each group seperate by newline are elf's calories
return the elf with the highest amount of calories

*** solution ***:
read the file

save the file into an array of characters
    - remove whitespace that may be in the saved line

have a l pointer
while l pointer is less than the length of the savedFile
    elfCalories
    while currentLine is not ""
        elfCalories += calorioes
        l += 1
    maxCal = max(elfCalories, max)
print(maxCalories)
*/
use std::fs;

fn main() {
    println!("sanith check");
    let contents = fs::read_to_string("input.txt").expect("cannot find the file");

    println!("{:?}", contents);
}
